src/panels/developer-tools/statistics/developer-tools-statistics.ts

```diff
@@ -0,0 +1,156 @@
+import "../../../components/data-table/ha-data-table";
+import type { DataTableColumnContainer } from "../../../components/data-table/ha-data-table";
+import { showAlertDialog } from "../../../dialogs/generic/show-dialog-box";
+import { showFixStatisticsUnitsChangedDialog } from "./show-dialog-statistics-fix-units-changed";
+
+@customElement("developer-tools-statistics")
+class HaPanelDevStatistics extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @state() private _data: (StatisticsValidationResult["data"] & {
+    type: string;
+  })[] = [];
+
+  protected firstUpdated() {
+    this._validateStatistics();
+  }
+
+  private _columns: DataTableColumnContainer = {
+    statistic_id: {
+      title: "Entity",
+      sortable: true,
+      filterable: true,
+      direction: "asc",
+      grows: true,
+    },
+    type: {
+      title: "Issue",
+      sortable: true,
+      filterable: true,
+      direction: "asc",
+      width: "60%",
+      template: (type, data) =>
+        html`${this.hass.localize(
+          `ui.panel.developer-tools.tabs.statistics.issues.${type}`,
+          data
+        ) || type}`,
+    },
+    fix: {
+      title: "",
+      template: (_, data) =>
+        html`<mwc-button @click=${this._fixIssue} .data=${data}
+          >Fix issue</mwc-button
+        >`,
+      width: "113px",
+    },
+  };
+
+  protected render() {
+    return html`
+      <ha-data-table
+        .columns=${this._columns}
+        .data=${this._data}
+        noDataText="No issues found!"
+      ></ha-data-table>
+    `;
+  }
+
+  private async _validateStatistics() {
+    const issues = Object.values(await validateStatistics(this.hass));
+    this._data = [];
+    issues.forEach((results) => {
+      results.forEach((issue) => {
+        this._data?.push({ ...issue.data, type: issue.type });
+      });
+    });
+  }
+
+  private _fixIssue(ev) {
+    const issue = ev.currentTarget.data;
+    if (issue.type === "unsupported_unit") {
+      showAlertDialog(this, {
+        title: "Unsupported unit",
+        text: html`The unit of your entity is not a suppported unit for the
+          device class of the entity, ${issue.device_class}. <br />Statistics
+          can not be generated until this entity has a supported unit.
+          <br /><br />If this unit was provided by an integration, this is a
+          bug. Please report an issue. <br /><br />If you have set this unit
+          yourself, and want to have statistics generated, make sure the unit
+          matched the device class. The supported units are documented in the
+          <a
+            href="https://developers.home-assistant.io/docs/core/entity/sensor"
+            target="_blank"
+          >
+            developer documentation</a
+          >.`,
+      });
+      return;
+    }
+    if (issue.type === "units_changed") {
+      showFixStatisticsUnitsChangedDialog(this, issue);
```

Comment:
Should we reload this page when it's fixed?

Reply:
Oh yeah! That was on my to-do still... Forgot it